runtime/nsd_ossl.c

```diff
@@ -633,6 +644,9 @@ osslChkPeerFingerprint(nsd_ossl_t *pThis, X509 *pCert)
 		if(!rsCStrSzStrCmp(pstrFingerprint, pPeer->pszID, strlen((char*) pPeer->pszID))) {
 			dbgprintf("osslChkPeerFingerprint: peer's certificate MATCH found: %s\n", pPeer->pszID);
 			bFoundPositiveMatch = 1;
+		} else if(!rsCStrSzStrCmp(pstrFingerprintSha256, pPeer->pszID, strlen((char*) pPeer->pszID))) {
+			dbgprintf("osslChkPeerFingerprint: peer's certificate MATCH found: %s\n", pPeer->pszID);
```

Comment:
Perhaps adding "Sha256" and "Sha1" to dbgprintf() ? would help to debug which matched the fingerprint